keras_cv/src/models/object_detection/faster_rcnn/feature_pyramid.py

```diff
@@ -0,0 +1,244 @@
+
+
+@keras_cv_export(
+    "keras_cv.models.faster_rcnn.FeaturePyramid",
+    package="keras_cv.models.faster_rcnn",
+)
+class FeaturePyramid(keras.layers.Layer):
+    """Implements a Feature Pyramid Network.
+
+    This implements the paper:
+      Tsung-Yi Lin, Piotr Dollar, Ross Girshick, Kaiming He, Bharath Hariharan,
+      and Serge Belongie. Feature Pyramid Networks for Object Detection.
+      (https://arxiv.org/pdf/1612.03144)
+
+    Feature Pyramid Networks (FPNs) are basic components that are added to an
+    existing feature extractor (CNN) to combine features at different scales.
+    For the basic FPN, the inputs are features `Ci` from different levels of a
+    CNN, which is usually the last block for each level, where the feature is
+    scaled from the image by a factor of `1/2^i`.
+
+    There is an output associated with each level in the basic FPN. The output
+    Pi at level `i` (corresponding to Ci) is given by performing a merge
+    operation on the outputs of:
+
+    1) a lateral operation on Ci (usually a conv2D layer with kernel = 1 and
+       strides = 1)
+    2) a top-down upsampling operation from Pi+1 (except for the top most level)
+
+    The final output of each level will also have a conv2D operation
+    (typically with kernel = 3 and strides = 1).
+
+    The inputs to the layer should be a dict with int keys should match the
+    pyramid_levels, e.g. for `pyramid_levels` = [2,3,4,5], the expected input
+    dict should be `{2:c2, 3:c3, 4:c4, 5:c5}`.
+
+    The output of the layer will have same structures as the inputs, a dict with
+    int keys and value for each of the level.
+
+    Args:
+        min_level: a python int for the lowest level of the pyramid for
+            feature extraction.
+        max_level: a python int for the highest level of the pyramid for
+            feature extraction.
+        num_channels: an integer representing the number of channels for the FPN
+            operations, defaults to 256.
+        lateral_layers: a python dict with int keys that matches to each of the
+            pyramid level. The values of the dict should be `keras.Layer`, which
+            will be called with feature activation outputs from backbone at each
+            level. Defaults to None, and a `keras.Conv2D` layer with kernel 1x1
+            will be created for each pyramid level.
+        output_layers: a python dict with int keys that matches to each of the
+            pyramid level. The values of the dict should be `keras.Layer`, which
+            will be called with feature inputs and merged result from upstream
+            levels. Defaults to None, and a `keras.Conv2D` layer with kernel 3x3
+            will be created for each pyramid level.
+
+    Example:
+    ```python
+    images = keras.layers.Input(
+        image_shape,
+        name="images",
+    )
+    extractor_levels= ["P2", "P3", "P4", "P5"]
+
+    backbone = keras_cv.models.ResNetV2Backbone.from_preset(
```

Comment:
make sure that the example in the docstring will run without errors

Reply:
Done